02_Util/src/util/swagger.ts

```diff
@@ -117,4 +125,126 @@ export function initSwagger(
   }
 
   server.register(fastifySwaggerUi, swaggerUiOptions);
+};
+
+export const convertHeadersToMap = (
+  headers: string[],
+): Record<string, string> => {
+  const headersMap: Record<string, string> = {};
+
+  for (let i = 0; i < headers.length; i += 2) {
+    const key = headers[i];
+    const value = headers[i + 1];
+    headersMap[key] = value;
+  }
+
+  return headersMap;
+};
+
+export const getHeaderValue = (
+  headers: string[],
+  key: string,
+): string | undefined => {
+  for (let i = 0; i < headers.length; i += 2) {
+    if (headers[i].toLowerCase() === key.toLowerCase()) {
+      return headers[i + 1];
+    }
+  }
+  return undefined;
+};
+
+const getTokenFromAuthHeader = (
+  authorizationHeader: string | undefined,
+): string | undefined => {
+  if (!!authorizationHeader) {
+    const token = authorizationHeader.split('Bearer ')[1];
+    return token;
+  }
+  return undefined;
+};
+
+const getAuthorizationTokenFromRawHeaders = (
+  headers: string[],
+): string | undefined => {
+  const authorizationHeader = getHeaderValue(headers, HttpHeader.Authorization);
+  return getTokenFromAuthHeader(authorizationHeader);
+};
+
+export const getAuthorizationTokenFromRequest = (
+  request: FastifyRequest,
+): string => {
+  const token = getAuthorizationTokenFromRawHeaders(request.raw.rawHeaders);
+  if (!token) {
+    throw new UnauthorizedException('Token not found in headers');
+  }
+  return token;
+};
+
+const registerFastifyAuth = async (server: FastifyInstance) => {
+  await server.register(FastifyAuth as any).after();
+  console.log((server as any).authorization);
+
+  server.decorate(
```

Comment:
This decorator callback function will run and authorize the request. This is the way for fastify to have authorization headers available for parsing. Currently left token hardcoded, but will look to implement logic that checks token against credentials module